students/MikeSchincariol/session5/file_copy.py

```diff
@@ -0,0 +1,27 @@
+import os.path
+
+def file_copy(src, dst):
+
+    CHUNK_SIZE = 128
+
+    # Open the files for reading and writing. Don't try
+    # and catch any errors as we want the caller to get the exception
+    # and deal with it.
+    num_bytes_copied = 0
+    with open(src, "rb") as src_fd:
+        with open(dst, "wb") as dst_fd:
+            while True:
+                # Read in at most CHUNK_SIZE bytes and then write them
+                # to the output stream. Keep track of how many byes
+                # have been transferred.
+                buf = src_fd.read(CHUNK_SIZE)
+                if len(buf) == 0:
+                    break
+                else:
+                    num_bytes_copied += len(buf)
+                    dst_fd.write(buf)
+    print("{0} bytes copied".format(num_bytes_copied))
+
+
+if __name__ == "__main__":
+    file_copy("students.txt", "students_copy.txt")
```

Comment:
Hi-

Sometimes this is overkill, but in instances where I'm looping through a
directory and passing several files to a function, it can be helpful to
have something that generates a new output file name instead of hard-coding
it. Your code looks great, so I was just trying to think of ideas that
might make it more efficient.. totally unnecessary in this case, however.

Summer

On Mon, Feb 15, 2016 at 6:21 PM, MikeSchincariol <notifications@github.com>
wrote:

> In students/MikeSchincariol/session5/file_copy.py
> <https://github.com/UWPCE-PythonCert/IntroPython2016a/pull/71#discussion_r52962505>
> :
>
> > +        with open(dst, "wb") as dst_fd:
> > +            while True:
> > +                # Read in at most CHUNK_SIZE bytes and then write them
> > +                # to the output stream. Keep track of how many byes
> > +                # have been transferred.
> > +                buf = src_fd.read(CHUNK_SIZE)
> > +                if len(buf) == 0:
> > +                    break
> > +                else:
> > +                    num_bytes_copied += len(buf)
> > +                    dst_fd.write(buf)
> > +    print("{0} bytes copied".format(num_bytes_copied))
> > +
> > +
> > +if __name__ == "__main__":
> > +    file_copy("students.txt", "students_copy.txt")
>
> Can you elabotate....is the new argument itself a function for generating
> the output file name?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/UWPCE-PythonCert/IntroPython2016a/pull/71/files#r52962505>
> .
>



-- 
Thanks,
Summer Rae
summerrae78@gmail.com